features/topology-map/org.opennms.features.topology.api/src/main/java/org/opennms/features/topology/api/browsers/OnmsVaadinContainer.java

```diff
@@ -184,7 +184,7 @@ public void reload(Page page) {
             }
             // Ensure that the number of items expected matches with the actual ones. See issue NMS-8079 fore more details.
             if (beans.size() != rowMap.size()) {
-                throw new IllegalStateException("The cache is supposed to carry " + beans.size() + " but only contains " + rowMap.size() + " items.");
+                throw new IllegalStateException("The cache is supposed to carry " + beans.size() + " but contains " + rowMap.size() + " items.");
```

Comment:
In this case, it would of been useful if the exception also carried extra details about the contents of the bean and the resulting map.

Reply:
In this case we would have get 2 lists. One containing 23, and the other one containing 22 items. A manual debugging/investigation of the problem is still needed. In my opinion it is not very helpful to show that information. This exception is purely to raise awareness that something is wrong.